src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -149,11 +150,32 @@ class RestZenStoreConfiguration(StoreConfiguration):
     """
 
     type: StoreType = StoreType.REST
-    username: str
-    password: str = ""
+    username: Optional[str] = None
+    password: Optional[str] = None
+    api_token: Optional[str] = None
     verify_ssl: Union[bool, str] = True
     http_timeout: int = DEFAULT_HTTP_TIMEOUT
 
+    @root_validator
+    def validate_credentials(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validates the credentials provided in the values dictionary.
+
+        Args:
+            values: A dictionary containing the values to be validated.
+
+        Raises:
+            ValueError: If neither api_token nor username is set.
+
+        Returns:
+            The values dictionary.
+        """
+        # Check if the values dictionary contains either an api_token or a username
```

Comment:
I had a few cases where this `values` dict actually contains lots of key-value pairs, but all the values are `None`.

If we want the token or username to be configured to an actual value, I'd recommend something like this:
```python
if values.get("api_token") is None and values.get("username") is None:
  ...

# the previous version allows empty strings, the requires non-empty strings
if values.get("api_token") and values.get("username"):
  ...
```

Reply:
@schustmi  sorry, i'm a bit confused about this,
We should check if at least one of them is not an empty string. does that what the following code do?
```if values.get("api_token") and values.get("username"):```

Reply:
Yes exactly!